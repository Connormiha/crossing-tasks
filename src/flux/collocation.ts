import schema from 'reducers/schema';
import immutable from 'immutability-helper';
import {RIVERSIDE_LEFT, RIVERSIDE_RIGHT, BOAT, ICollocationState} from 'flux/types';

import {
    COLLOCATION_MOVE_CHARACTER, COLLOCATION_MOVE_BOAT, COLLOCATION_INIT, COLLOCATION_TOGGLE_BOAT_INVALID,
} from './constants';

/**
 * Moves character to other position (form/to boat)
 */
export const moveCharacter = (id: string) =>
    ({type: COLLOCATION_MOVE_CHARACTER, id});

/**
 * Move boat to other riverside
 */
export const moveBoat = () =>
    ({type: COLLOCATION_MOVE_BOAT});

/**
 * Move boat to other riverside
 */
export const toggleBoatInvalid = (isBoatInvalid: boolean) =>
    ({type: COLLOCATION_TOGGLE_BOAT_INVALID, isBoatInvalid});

/**
 * Init new collocation
 */
export const init = (collocation: any) =>
    ({type: COLLOCATION_INIT, collocation});

const getDefaultState = (): ICollocationState =>
    schema.collocation;

export default (state: ICollocationState = getDefaultState(), {type, collocation, isBoatInvalid, id}): ICollocationState => {
    switch (type) {
        case COLLOCATION_MOVE_CHARACTER:
            for (const item of [RIVERSIDE_LEFT, RIVERSIDE_RIGHT, BOAT]) {
                const index: number = state[item].indexOf(id);

                if (index !== -1) {
                    const moveTo = item === BOAT ? state.boatPosition : BOAT;

                    return immutable(
                        state,
                        {
                            [item]: {
                                $set: [...state[item].slice(0, index), ...state[item].slice(index + 1)],
                            },
                            [moveTo]: {
                                $set: state[moveTo].concat(id),
                            },
                        },
                    );
                }
            }

            return state;

        case COLLOCATION_MOVE_BOAT:
            return immutable(state, {boatPosition: {$set: state.boatPosition === RIVERSIDE_LEFT ? RIVERSIDE_RIGHT : RIVERSIDE_LEFT}});

        case COLLOCATION_TOGGLE_BOAT_INVALID:
            return immutable(state, {isBoatInvalid: {$set: isBoatInvalid}});

        case COLLOCATION_INIT:
            return immutable(state, {$merge: {...collocation, boatPosition: RIVERSIDE_LEFT}});
    }

    return state;
};
