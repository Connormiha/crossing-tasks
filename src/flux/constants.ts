export const COLLOCATION_MOVE_CHARACTER: string = 'COLLOCATION_MOVE_CHARACTER';
export const COLLOCATION_MOVE_BOAT: string = 'COLLOCATION_MOVE_BOAT';
export const COLLOCATION_INIT: string = 'COLLOCATION_INIT';

export const GAME_SET: string = 'GAME_SET';
export const GAME_INIT: string = 'GAME_INIT';

export const MESSEGE_SET: string = 'MESSEGE_SET';
