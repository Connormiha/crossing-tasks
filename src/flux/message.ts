import schema from 'reducers/schema';

const MESSEGE_SET: string = 'MESSEGE_SET';

/**
 * Sets message
 */
export const set = (content: string) =>
    ({type: MESSEGE_SET, content});

const getDefaultState = () =>
    immutable(schema.message);

export default (state = getDefaultState(), {type, content}) => {
    switch (type) {
        case MESSEGE_SET:
            return state.set('content', content);
    }

    return state;
};
