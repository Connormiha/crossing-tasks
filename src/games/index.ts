import game1 from 'games/game1';
import game2 from 'games/game2';
import game3 from 'games/game3';
import game4 from 'games/game4';
import game5 from 'games/game5';
import game6 from 'games/game6';
import game7 from 'games/game7';

import {
    Game, ValidatorResult, PositionCharacter
} from 'games/helpers';

export {
    Game, ValidatorResult, PositionCharacter
};

const games = {
    game1, game2, game3, game4, game5, game6, game7
};

export default games;
