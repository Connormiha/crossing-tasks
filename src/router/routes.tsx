import * as React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import PagePlay from 'components/page/play';
import PageEntry from 'components/page/entry';
import NotFound from 'components/page/not-found';
import {ROOT_URL} from 'config/config';

export default (
    <Router basename={`/${ROOT_URL}`}>
        <Switch>
            <Route path="/" exact component={PageEntry} />
            <Route path="/play/:id/" exact component={PagePlay} />
            <Route path="*" component={NotFound} />
        </Switch>
    </Router>
);
