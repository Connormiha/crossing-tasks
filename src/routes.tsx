/* tslint:disable:no-unused-variable */
import React from 'react';
/* tslint:enable:no-unused-variable */

import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import PagePlay from 'components/page/Play';
import PageEntry from 'components/page/Entry';
import NotFound from 'components/page/NotFound';

export default (
    <Router basename={`/${process.env.ROOT_URL}`}>
        <Switch>
            <Route path="/" exact component={PageEntry} />
            <Route path="/play/:id/" exact component={PagePlay} />
            <Route path="*" component={NotFound} />
        </Switch>
    </Router>
);
