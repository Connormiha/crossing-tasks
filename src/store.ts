import reducers from './reducers';
import { createStore, Store } from 'redux';
import { enableBatching } from 'redux-batched-actions';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

export const createAppStore = (): Store =>
  createStore(enableBatching(reducers), composeWithDevTools());

export default createAppStore();
