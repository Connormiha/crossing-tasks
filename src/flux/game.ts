import schema from 'reducers/schema';
import immutable from 'immutability-helper';

import {
    GAME_SET, GAME_INIT, GAME_FINISH
} from './constants';

import {IGameState} from 'flux/types';

/**
 * Sets currentGame
 */
export const set = (id: string) =>
    ({type: GAME_SET, id});

export const init = (list: string[]) =>
    ({type: GAME_INIT, list});

export const finish = () =>
    ({type: GAME_FINISH});

const getDefaultState = (): IGameState =>
    schema.game;

export default (state: IGameState = getDefaultState(), {type, id, list}): IGameState => {
    switch (type) {
        case GAME_SET:
            return immutable(state, {$merge: {currentGame: id, finished: false}});

        case GAME_INIT:
            return immutable(state, {list: {$set: list}});

        case GAME_FINISH:
            return immutable(state, {finished: {$set: true}});
    }

    return state;
};
