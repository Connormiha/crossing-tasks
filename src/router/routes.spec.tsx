import * as renderer from 'react-test-renderer';
import {createAppStore} from 'store';
import {Provider} from 'react-redux';
import routes from 'router/routes';
import games from 'games';

// Redux
import * as gameActions from 'flux/game';

/* tslint:disable:jsx-wrap-multiline */
describe('Route', () => {
    const originalHref = location.href;
    const originalTitle = document.title;
    let store;

    beforeEach(() => {
        store = createAppStore();
        store.dispatch(gameActions.init(Object.keys(games)));
    });

    afterAll(() => {
        global.history.replaceState({}, originalTitle, originalHref);
    });

    describe('<EntryPage />', () => {
        it('should render', () => {
            global.history.replaceState({}, originalTitle, '/');

            const tree = renderer.create(
                <Provider store={store}>
                    {routes}
                </Provider>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('<PlayPage />', () => {
        for (let i = 1; i < 7; i++) {
            it(`should render game_${i}`, () => {
                global.history.replaceState({}, originalTitle, `/play/game_${i}/`);

                const tree = renderer.create(
                    <Provider store={store}>
                        {routes}
                    </Provider>
                ).toJSON();

                expect(tree).toMatchSnapshot();
            });
        }
    });

    describe('<NotFoundPage />', () => {
        it(`should render /foo`, () => {
            global.history.replaceState({}, originalTitle, '/foo');

            const tree = renderer.create(
                <Provider store={store}>
                    {routes}
                </Provider>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
