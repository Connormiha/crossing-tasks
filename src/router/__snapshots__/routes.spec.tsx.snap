// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route <EntryPage /> should render 1`] = `
<div
  className="page-entry"
>
  <div
    className="page-entry__content"
  >
    <ul
      className="list-games"
    >
      <li
        className="list-games__item"
      >
        <a
          href="/play/game1/"
          onClick={[Function]}
        >
          game1
        </a>
      </li>
      <li
        className="list-games__item"
      >
        <a
          href="/play/game2/"
          onClick={[Function]}
        >
          game2
        </a>
      </li>
      <li
        className="list-games__item"
      >
        <a
          href="/play/game3/"
          onClick={[Function]}
        >
          game3
        </a>
      </li>
      <li
        className="list-games__item"
      >
        <a
          href="/play/game4/"
          onClick={[Function]}
        >
          game4
        </a>
      </li>
      <li
        className="list-games__item"
      >
        <a
          href="/play/game5/"
          onClick={[Function]}
        >
          game5
        </a>
      </li>
      <li
        className="list-games__item"
      >
        <a
          href="/play/game6/"
          onClick={[Function]}
        >
          game6
        </a>
      </li>
      <li
        className="list-games__item"
      >
        <a
          href="/play/game7/"
          onClick={[Function]}
        >
          game7
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Route <NotFoundPage /> should render /foo 1`] = `
<div
  className="page-not-found"
>
  <h1
    className="page-not-found__title"
  >
    Page not found
  </h1>
</div>
`;

exports[`Route <PlayPage /> should render game1 1`] = `
<div
  className="page-play"
>
  <div
    className="page-play__content"
  >
    <div
      className="riverside riverside_side_left"
    >
      <div>
        <button
          className="character character_skin_farmer"
          onClick={[Function]}
          title="Farmer"
        />
      </div>
      <div>
        <button
          className="character character_skin_cabbage"
          onClick={[Function]}
          title="Cabbage"
        />
      </div>
      <div>
        <button
          className="character character_skin_sheep"
          onClick={[Function]}
          title="Sheep"
        />
      </div>
      <div>
        <button
          className="character character_skin_wolf"
          onClick={[Function]}
          title="Wolf"
        />
      </div>
    </div>
    <div
      className="riverside riverside_side_right"
    />
    <div
      className="boat boat_position_left"
      onAnimationEnd={[Function]}
      onTransitionEnd={[Function]}
    />
    <button
      className="remote"
      disabled={true}
      onClick={[Function]}
    >
      Go
    </button>
    <div
      className="settings"
    >
      <div
        className="settings__volume settings__volume_type_high"
      >
        <button
          className="settings__volume-toggler"
          onClick={[Function]}
        />
        <input
          className="settings__volume-range settings__volume-range_hidden"
          max="1"
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          step="0.05"
          type="range"
          value={1}
        />
      </div>
    </div>
    <audio>
      <source
        src="assets/audio/wave.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/wave.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/water-drop.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/water-drop.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/shaker.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/shaker.mp3"
        type="audio/mpeg"
      />
    </audio>
    
  </div>
</div>
`;

exports[`Route <PlayPage /> should render game2 1`] = `
<div
  className="page-play"
>
  <div
    className="page-play__content"
  >
    <div
      className="riverside riverside_side_left"
    >
      <div>
        <button
          className="character character_skin_caveman"
          onClick={[Function]}
          title="Caveman"
        />
      </div>
      <div>
        <button
          className="character character_skin_caveman"
          onClick={[Function]}
          title="Caveman"
        />
      </div>
      <div>
        <button
          className="character character_skin_caveman"
          onClick={[Function]}
          title="Caveman"
        />
      </div>
      <div>
        <button
          className="character character_skin_priest"
          onClick={[Function]}
          title="Priest"
        />
      </div>
      <div>
        <button
          className="character character_skin_priest"
          onClick={[Function]}
          title="Priest"
        />
      </div>
      <div>
        <button
          className="character character_skin_priest"
          onClick={[Function]}
          title="Priest"
        />
      </div>
    </div>
    <div
      className="riverside riverside_side_right"
    />
    <div
      className="boat boat_position_left"
      onAnimationEnd={[Function]}
      onTransitionEnd={[Function]}
    />
    <button
      className="remote"
      disabled={true}
      onClick={[Function]}
    >
      Go
    </button>
    <div
      className="settings"
    >
      <div
        className="settings__volume settings__volume_type_high"
      >
        <button
          className="settings__volume-toggler"
          onClick={[Function]}
        />
        <input
          className="settings__volume-range settings__volume-range_hidden"
          max="1"
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          step="0.05"
          type="range"
          value={1}
        />
      </div>
    </div>
    <audio>
      <source
        src="assets/audio/wave.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/wave.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/water-drop.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/water-drop.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/shaker.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/shaker.mp3"
        type="audio/mpeg"
      />
    </audio>
    
  </div>
</div>
`;

exports[`Route <PlayPage /> should render game3 1`] = `
<div
  className="page-play"
>
  <div
    className="page-play__content"
  >
    <div
      className="riverside riverside_side_left"
    >
      <div>
        <button
          className="character character_skin_woman_red"
          onClick={[Function]}
          title="Woman_red"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_red"
          onClick={[Function]}
          title="Men_red"
        />
      </div>
      <div>
        <button
          className="character character_skin_woman_white"
          onClick={[Function]}
          title="Woman_white"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_white"
          onClick={[Function]}
          title="Men_white"
        />
      </div>
      <div>
        <button
          className="character character_skin_woman_black"
          onClick={[Function]}
          title="Woman_black"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_black"
          onClick={[Function]}
          title="Men_black"
        />
      </div>
    </div>
    <div
      className="riverside riverside_side_right"
    />
    <div
      className="boat boat_position_left"
      onAnimationEnd={[Function]}
      onTransitionEnd={[Function]}
    />
    <button
      className="remote"
      disabled={true}
      onClick={[Function]}
    >
      Go
    </button>
    <div
      className="settings"
    >
      <div
        className="settings__volume settings__volume_type_high"
      >
        <button
          className="settings__volume-toggler"
          onClick={[Function]}
        />
        <input
          className="settings__volume-range settings__volume-range_hidden"
          max="1"
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          step="0.05"
          type="range"
          value={1}
        />
      </div>
    </div>
    <audio>
      <source
        src="assets/audio/wave.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/wave.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/water-drop.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/water-drop.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/shaker.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/shaker.mp3"
        type="audio/mpeg"
      />
    </audio>
    
  </div>
</div>
`;

exports[`Route <PlayPage /> should render game4 1`] = `
<div
  className="page-play"
>
  <div
    className="page-play__content"
  >
    <div
      className="riverside riverside_side_left"
    >
      <div>
        <button
          className="character character_skin_men_black"
          onClick={[Function]}
          title="Men_black"
        />
      </div>
      <div>
        <button
          className="character character_skin_boy_red"
          onClick={[Function]}
          title="Boy_red"
        />
      </div>
      <div>
        <button
          className="character character_skin_boy_yellow"
          onClick={[Function]}
          title="Boy_yellow"
        />
      </div>
      <div>
        <button
          className="character character_skin_woman_black"
          onClick={[Function]}
          title="Woman_black"
        />
      </div>
      <div>
        <button
          className="character character_skin_girl_red"
          onClick={[Function]}
          title="Girl_red"
        />
      </div>
      <div>
        <button
          className="character character_skin_girl_yellow"
          onClick={[Function]}
          title="Girl_yellow"
        />
      </div>
      <div>
        <button
          className="character character_skin_criminal"
          onClick={[Function]}
          title="Criminal"
        />
      </div>
      <div>
        <button
          className="character character_skin_policeman"
          onClick={[Function]}
          title="Policeman"
        />
      </div>
    </div>
    <div
      className="riverside riverside_side_right"
    />
    <div
      className="boat boat_position_left"
      onAnimationEnd={[Function]}
      onTransitionEnd={[Function]}
    />
    <button
      className="remote"
      disabled={true}
      onClick={[Function]}
    >
      Go
    </button>
    <div
      className="settings"
    >
      <div
        className="settings__volume settings__volume_type_high"
      >
        <button
          className="settings__volume-toggler"
          onClick={[Function]}
        />
        <input
          className="settings__volume-range settings__volume-range_hidden"
          max="1"
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          step="0.05"
          type="range"
          value={1}
        />
      </div>
    </div>
    <audio>
      <source
        src="assets/audio/wave.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/wave.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/water-drop.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/water-drop.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/shaker.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/shaker.mp3"
        type="audio/mpeg"
      />
    </audio>
    
  </div>
</div>
`;

exports[`Route <PlayPage /> should render game5 1`] = `
<div
  className="page-play"
>
  <div
    className="page-play__content"
  >
    <div
      className="riverside riverside_side_left"
    >
      <div>
        <button
          className="character character_skin_priest"
          onClick={[Function]}
          title="Priest"
        />
      </div>
      <div>
        <button
          className="character character_skin_priest"
          onClick={[Function]}
          title="Priest"
        />
      </div>
      <div>
        <button
          className="character character_skin_priest"
          onClick={[Function]}
          title="Priest"
        />
      </div>
      <div>
        <button
          className="character character_skin_monkey"
          onClick={[Function]}
          title="Monkey"
        />
      </div>
      <div>
        <button
          className="character character_skin_monkey"
          onClick={[Function]}
          title="Monkey"
        />
      </div>
      <div>
        <button
          className="character character_skin_gorilla"
          onClick={[Function]}
          title="Gorilla"
        />
      </div>
    </div>
    <div
      className="riverside riverside_side_right"
    />
    <div
      className="boat boat_position_left"
      onAnimationEnd={[Function]}
      onTransitionEnd={[Function]}
    />
    <button
      className="remote"
      disabled={true}
      onClick={[Function]}
    >
      Go
    </button>
    <div
      className="settings"
    >
      <div
        className="settings__volume settings__volume_type_high"
      >
        <button
          className="settings__volume-toggler"
          onClick={[Function]}
        />
        <input
          className="settings__volume-range settings__volume-range_hidden"
          max="1"
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          step="0.05"
          type="range"
          value={1}
        />
      </div>
    </div>
    <audio>
      <source
        src="assets/audio/wave.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/wave.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/water-drop.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/water-drop.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/shaker.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/shaker.mp3"
        type="audio/mpeg"
      />
    </audio>
    
  </div>
</div>
`;

exports[`Route <PlayPage /> should render game6 1`] = `
<div
  className="page-play"
>
  <div
    className="page-play__content"
  >
    <div
      className="riverside riverside_side_left"
    >
      <div>
        <button
          className="character character_skin_woman_red"
          onClick={[Function]}
          title="Woman_red"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_red"
          onClick={[Function]}
          title="Men_red"
        />
      </div>
      <div>
        <button
          className="character character_skin_woman_white"
          onClick={[Function]}
          title="Woman_white"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_white"
          onClick={[Function]}
          title="Men_white"
        />
      </div>
      <div>
        <button
          className="character character_skin_woman_black"
          onClick={[Function]}
          title="Woman_black"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_black"
          onClick={[Function]}
          title="Men_black"
        />
      </div>
      <div>
        <button
          className="character character_skin_woman_blue"
          onClick={[Function]}
          title="Woman_blue"
        />
      </div>
      <div>
        <button
          className="character character_skin_men_blue"
          onClick={[Function]}
          title="Men_blue"
        />
      </div>
    </div>
    <div
      className="riverside riverside_side_right"
    />
    <div
      className="boat boat_position_left"
      onAnimationEnd={[Function]}
      onTransitionEnd={[Function]}
    />
    <button
      className="remote"
      disabled={true}
      onClick={[Function]}
    >
      Go
    </button>
    <div
      className="settings"
    >
      <div
        className="settings__volume settings__volume_type_high"
      >
        <button
          className="settings__volume-toggler"
          onClick={[Function]}
        />
        <input
          className="settings__volume-range settings__volume-range_hidden"
          max="1"
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          step="0.05"
          type="range"
          value={1}
        />
      </div>
    </div>
    <audio>
      <source
        src="assets/audio/wave.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/wave.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/water-drop.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/water-drop.mp3"
        type="audio/mpeg"
      />
    </audio>
    <audio>
      <source
        src="assets/audio/shaker.opus"
        type="audio/ogg; codecs=opus"
      />
      <source
        src="assets/audio/shaker.mp3"
        type="audio/mpeg"
      />
    </audio>
    
  </div>
</div>
`;
